### PR TITLE
Added the background particle effects

### LINKED ISSUE #1227: Finish the particle effects
Some particle effect work was done by @DonGororo but it needs cleaning up and making it work all nice by a programmer.
That's on this branch: https://github.com/Revolutionary-Games/Thrive/tree/particles

### PR DESCRIPTION
**Pull request by Kasterisk**
Made the background particles spawn and changes according to the biome (background). It's currently a child of the camera and follows the camera's position.
Closes #1227

### CODE REVIEW

File: assets/textures/New_canvas.png.import [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[remap]
```
Comment (hhyyrylainen) [inline assets/textures/New_canvas.png.import:1]: This file should have a more descriptive name

File: assets/textures/bubble 2.png.import [hunk 2]
```diff
@@ -0,0 +1,34 @@
+[remap]
```
Comment (hhyyrylainen) [inline assets/textures/bubble 2.png.import:1]: filename should not have a space

File: assets/textures/particle.png.import [hunk 3]
```diff
@@ -0,0 +1,34 @@
+[remap]
```
Comment (hhyyrylainen) [inline assets/textures/particle.png.import:1]: this should have a more descriptive name

File: assets/textures/particlefuzzy.png.import [hunk 4]
```diff
@@ -0,0 +1,36 @@
+[remap]
```
Comment (hhyyrylainen) [inline assets/textures/particlefuzzy.png.import:1]: same with this

File: assets/textures/shard.png.import [hunk 5]
```diff
@@ -0,0 +1,36 @@
+[remap]
```
Comment (hhyyrylainen) [inline assets/textures/shard.png.import:1]: `shard_particle.png`

File: src/microbe_stage/AgentProjectile2.tscn [hunk 6]
```diff
@@ -0,0 +1,64 @@
+[gd_scene load_steps=11 format=2]
```
Comment (hhyyrylainen) [inline src/microbe_stage/AgentProjectile2.tscn:1]: This doesn't seem to be used anywhere?
I don't think an unused scene should be included in the main branch unless this is moved somewhere like `test/experimental_scenes`

File: src/microbe_stage/MicrobeCamera.cs [hunk 7]
```diff
@@ -144,6 +146,16 @@ public void SetBackground(Background background)
         {
             materialToUpdate.SetShaderParam($"layer{i:n0}", GD.Load<Texture>(background.Textures[i]));
         }
+
+        if (BackgroundParticles != null)
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeCamera.cs:150]: I don't particularly like that the particles are used in the editor.
Making an [Export] bool variable in the camera class that disables them would be nice, and then setting that true in the editor scene for the camera. (the same way the default zoom is different).

File: src/microbe_stage/particles/Ambientalspatialmaterial.tres [hunk 8]
```diff
@@ -0,0 +1,15 @@
+[gd_resource type="SpatialMaterial" load_steps=2 format=2]
```
Comment (hhyyrylainen) [inline src/microbe_stage/particles/Ambientalspatialmaterial.tres:1]: This file should be renamed with PascalCase

File: src/microbe_stage/particles/background/Current.tscn [hunk 9]
```diff
@@ -0,0 +1,59 @@
+[gd_scene load_steps=9 format=2]
```
Comment (hhyyrylainen) [inline src/microbe_stage/particles/background/Current.tscn:1]: I think there are also unused scenes in the particles and particles/background folders that should be moved to that experimental folder I mentioned before.

File: src/microbe_stage/particles/Ambientalspatialmaterial.tres (hunk 8 above)
Comment (Kasterisk) [reply src/microbe_stage/particles/Ambientalspatialmaterial.tres:1]: Hmmm, I tried to rename this file but it doesn't show up on the git changes to add

Comment (hhyyrylainen) [reply src/microbe_stage/particles/Ambientalspatialmaterial.tres:1]: yeah. that's a windows problem. I'll rename it.

**Comment by Kasterisk** [comment]
Couldn't find other problems (besides the unused AmbientalSpatialMaterial.tres). everything is good.

**Pull request merged**